src/main/java/gregtech/api/unification/material/properties/FluidProperty.java

```diff
@@ -0,0 +1,95 @@
+package gregtech.api.unification.material.properties;
+
+import com.google.common.base.Preconditions;
+import crafttweaker.api.liquid.ILiquidDefinition;
+import crafttweaker.api.minecraft.CraftTweakerMC;
+import gregtech.api.GTValues;
+import net.minecraftforge.fluids.Fluid;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fml.common.Optional;
+
+import javax.annotation.Nonnull;
+
+//@ZenClass("mods.gregtech.material.FluidMaterial")
+//@ZenRegister
+public class FluidProperty implements IMaterialProperty<FluidProperty> {
+
+    public static final int BASE_TEMP = 300;
+
+    /**
+     * Internal material fluid field
+     */
+    private Fluid fluid;
+
+    private boolean hasBlock;
+    private boolean isGas;
+    private int fluidTemperature = BASE_TEMP;
+
+    public FluidProperty(boolean isGas, boolean hasBlock) {
+        this.isGas = isGas;
+        this.hasBlock = hasBlock;
+    }
+
+    /**
+     * Default values of: no Block, not Gas.
+     */
+    public FluidProperty() {
+        this(false, false);
```

Comment:
I think we should default to adding FluidBlocks, as FluidBlocks are required for the Fluid Spring generator type in world generation.

Reply:
I disagree, as only a very select fluids previously had Fluid Blocks